sdk/search/azure-search/src/main/java/com/azure/search/SearchIndexAsyncClient.java

```diff
@@ -0,0 +1,736 @@
+     */
+    public Mono<Response<Document>> getDocumentWithResponse(
+        String key,
+        List<String> selectedFields,
+        RequestOptions requestOptions) {
+        return withContext(context -> getDocumentWithResponse(key, selectedFields, requestOptions, context));
+    }
+
+    Mono<Response<Document>> getDocumentWithResponse(
+        String key,
+        List<String> selectedFields,
+        RequestOptions requestOptions,
+        Context context) {
+        return restClient
+            .documents()
+            .getWithRestResponseAsync(key, selectedFields, requestOptions, context)
+            .map(res -> {
+                Document doc = res.getValue();
+                DocumentResponseConversions.cleanupDocument(doc);
+                return new SimpleResponse<>(res, doc);
+            })
+            .onErrorMap(DocumentResponseConversions::exceptionMapper)
+            .map(Function.identity());
+    }
+
+    /**
+     * Suggests documents in the Azure Cognitive Search index that match the given partial query text.
+     *
+     * @param searchText search text
+     * @param suggesterName suggester name
+     * @return A {@link PagedFluxBase} that iterates over {@link SuggestResult} objects
+     * and provides access to the {@link SuggestPagedResponse} object for each page containing
+     * HTTP response and coverage information.
+     */
+    public PagedFluxBase<SuggestResult, SuggestPagedResponse> suggest(String searchText, String suggesterName) {
+        return this.suggest(searchText, suggesterName, null, null);
+    }
+
+    /**
+     * Suggests documents in the Azure Cognitive Search index that match the given partial query text.
+     *
+     * @param searchText search text
+     * @param suggesterName suggester name
+     * @param suggestOptions suggest options
+     * @param requestOptions additional parameters for the operation.
+     * Contains the tracking ID sent with the request to help with debugging
+     * @return A {@link PagedFluxBase} that iterates over {@link SuggestResult} objects
+     * and provides access to the {@link SuggestPagedResponse} object for each page containing
+     * HTTP response and coverage information.
+     */
+    public PagedFluxBase<SuggestResult, SuggestPagedResponse> suggest(String searchText,
+                                                                      String suggesterName,
+                                                                      SuggestOptions suggestOptions,
+                                                                      RequestOptions requestOptions) {
+        SuggestRequest suggestRequest = this.createSuggestRequest(searchText,
+            suggesterName, SuggestOptionsHandler.ensureSuggestOptions(suggestOptions));
+        return new PagedFluxBase<>(
+            () -> withContext(context -> this.suggestFirst(requestOptions, suggestRequest, context)),
+            nextLink -> Mono.empty());
```

Comment:
Could this use the constructor that just takes a `firstPageRetriever` and implicitly uses `Mono.empty` for the next page value.